bicep/2025-06-manifest-to-bicep-extension-cli.md

```diff
@@ -0,0 +1,193 @@
+https://github.com/radius-project/radius/issues/9450
+
+### Goals
+
+- Remove the Node.js/NPM dependency from the `rad bicep publish-extension` command
+- Provide a consistent binary management experience similar to the Bicep CLI
+- Maintain backward compatibility with existing workflows
+- Ensure the binary is automatically downloaded when needed
+
+### Non goals
+
+- Changing the functionality of the manifest-to-bicep-extension tool itself
+- Modifying the Bicep extension format or publishing process
+- Supporting multiple versions of the manifest-to-bicep-extension tool
+
+### User scenarios
+
+#### User story 1
+As a platform engineer, I want to generate and publish Bicep extensions for my custom resource types without installing Node.js, so I can use Radius in environments where Node.js is not available. I can use `rad bicep download` to get the necessary binaries and then run `rad bicep publish-extension` to generate the Bicep extension from my resource provider manifest. I could also publish the `manifest-to-bicep-extension` binary to a central repository for my team to use, ensuring everyone has a consistent toolchain.
+
+## User Experience
+
+The user experience should be exactly the same as it is today, with the following changes:
+- The `rad bicep publish-extension` command will no longer require `npx` and will instead execute a native binary.
+- The user will need to run `rad bicep download` to download the new `manifest-to-bicep-extension` binary, similar to how they currently download the Bicep CLI.
+- The command will prompt the user to run `rad bicep download` if the binary is not found.
+
+## Design
+
+### High Level Design
+
+The solution replaces the `npx`-based approach with a native binary approach:
+
+1. The `rad bicep download` command is enhanced to download both the Bicep CLI and the `manifest-to-bicep-extension` binary
+2. The binaries are stored in the same directory (`~/.rad/bin/`)
+3. The `rad bicep publish-extension` command uses the downloaded binary instead of `npx`
+4. The binary download follows the same pattern as the Bicep CLI download
+
+### Architecture Diagram
+
+```
+┌─────────────────────┐
+│   rad bicep         │
+│   download          │
+└──────────┬──────────┘
+           │
+           ├─── Downloads Bicep CLI
+           │
+           └─── Downloads manifest-to-bicep-extension
+                           │
+                           v
+                   ┌───────────────┐
+                   │ ~/.rad/bin/   │
+                   │ ├── rad-bicep │
+                   │ └── manifest- │
+                   │     to-bicep- │
+                   │     extension │
+                   └───────────────┘
+```
+
+### Detailed Design
+
+#### Binary Management
+
+The manifest-to-bicep-extension binary is managed similarly to the Bicep CLI:
+
+1. **Download Location**: The binary is downloaded from GitHub releases at `https://github.com/radius-project/bicep-tools/releases/latest/download/`
```

Comment:
Is the binary something we develop, or does it exist somewhere else? If it's somewhere else, can you provide a link?